### PR TITLE
OS X: Use compliant directories for different resources

### PR DESCRIPTION
**Pull request by dajero**
This commit aims to make Meka's OS X version more compliant to the file system hierarchy used by the operating system. 

I have included the rest of the commit message below for your reference. I'd be interested in your feedback regarding my differentiation of the different types of resources. I am not too familiar with meka yet, so please correct me and let me know if other files require a special or different treatment too.

It does so by differentiating between three types of resources and placing them in suitable directories. These distinctions are made in file.c's Filenames_Init() function.

Makefile has been altered to handle osxhelpers.mm, an Objective-C++ file, that interfaces with OS X's API's. The client (file.c) is still compiled as usual.

First there are internal, read-only files, these include:
- meka.nam
- meka.dat
- meka.pat
- meka.blt
- meka.msg
- *.txt files
There are read from the Application Bundle's Resources directory

Then there are interal, writable files, which include
- meka.inp
- meka.dsk
- meka.cfg
- meka.fdb

These are written to ~/Library/Application\ Support/org.smspower.meka. Files that ship with meka (for now just meka.inp), should (and are) first copied to this directory.

Finally there are resources that a user will most likely create and want to access easily. These are placed in the Screenshots, Saves, Music and Debug directories which Meka will now create in ~/Documents/Meka.

### CODE REVIEW

File: meka/srcs/file.c [hunk 1]
```diff
@@ -116,43 +112,64 @@ void    Filenames_Init()
...
+		"meka.inp",
+		NULL
+	};
+	PopulateWritableInternalResourcesPath(internalResourcesDirectory, filesToCopyToInternalResourcesPath);
+#else
+	strcpy(internalResourcesDirectory, g_env.Paths.EmulatorDirectory);
+	strcpy(externalResourcesDirectory, g_env.Paths.EmulatorDirectory);
+#endif
+
+	ConsolePrintf("Data Directory = %s\n", dataDirectory);
```
Comment (ocornut) [inline meka/srcs/file.c]: Left over debug stuff.

**Comment by ocornut** [comment]
Sorry I left that unanswered, MEKA is unfortunately not at the top of my priority list at the moment and being not familiar with OSX ecosystem that sort of PR is rather alien to me.

When using MEKA - aka, an old software from a downloaded package - did it really bother you that those files were stored in the same directory?
I am enclined to merge this but not fancying too much adding new code that I can't really test either.

**Comment by ocornut** [comment]
Merged as a gesture of welcome :)

File: meka/srcs/file.c [hunk 2]
```diff
@@ -105,54 +104,72 @@ void    Filenames_Init()
...
 #endif
 
+	char dataDirectory[FILENAME_LEN], internalResourcesDirectory[FILENAME_LEN], externalResourcesDirectory[FILENAME_LEN];
+	strcpy(dataDirectory, g_env.Paths.EmulatorDirectory);
+#ifdef ARCH_MACOSX
+	GetWritableInternalResourcePath( internalResourcesDirectory, sizeof(internalResourcesDirectory) );
+	GetWritableExternalResourcePath( externalResourcesDirectory, sizeof(externalResourcesDirectory) );
+	const char *filesToCopyToInternalResourcesPath[2] = {
+		"meka.inp",
+		NULL
```
Comment (MikeDX) [inline meka/srcs/file.c]: This patch actually causes an issue if you are compiling a basic binary and not a bundled .app folder (MEKA crashes).

**Pull request merged**